### PR TITLE
Docker support for testing vim plugin

### PR DESCRIPTION
**Pull request by ccampanale**
- adds Dockerfile describing test image on alpine with curl, vim, git,
  the minimal configuration plugins, and an example .vimrc file
- adds an example.vimrc which is injected into the test image
- adds a bash script for easily building and running the docker test

### CODE REVIEW

**Comment by ccampanale** [comment]
So I started on what you mentioned. It's mostly working but there is an error getting the function to work... Not sure why just yet but I figured I toss this out.

`./test-vim-terraform-completion` will build/run the image opening `vim` against a new documentation `test.tf`.

**Comment by ccampanale** [comment]
Oh, and I added terraform to the image which makes it significantly larger as I wondered if that was why I was getting the error but it didn't help. Up to you if you want to keep it there for testing the linting stuff.

The error I'm getting right now is: `E117: Unknown function: terraformcomplete#Complete` when I try to use Omni completion. I'm a super novice with Vim so I'm sure you know exactly what the issue is. haha

**Comment by juliosueiras** [comment]
Currently reviewing it

**Comment by juliosueiras** [comment]
```Dockerfile
FROM base/archlinux
USER root
WORKDIR /root
ADD example.vimrc .vimrc
RUN pacman -Syy && pacman -S vim ruby git unzip --noconfirm
RUN curl -fLo ~/.vim/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim && vim +PlugInstall +qall
RUN curl https://releases.hashicorp.com/terraform/0.9.4/terraform_0.9.4_linux_amd64.zip -o terraform_0.9.4_linux_amd64.zip && unzip terraform_0.9.4_linux_amd64.zip && mv terraform /usr/bin/terraform
ENTRYPOINT ["vim"]
```

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,8 @@
+FROM alpine:3.5
+USER root
+WORKDIR /root
+ADD example.vimrc .vimrc
+RUN apk add --no-cache curl vim git terraform && \
+    curl -fLo ~/.vim/autoload/plug.vim --create-dirs https://raw.githubusercontent.com/junegunn/vim-plug/master/plug.vim && \
+    vim +PlugInstall +qall
```
Comment (juliosueiras) [inline Dockerfile]: Vim in alpine does not come with +ruby or +ruby/dyn , so the plugin won't work due to that

Comment (ccampanale) [inline Dockerfile]: Oh right... That makes sense.

**Comment by ccampanale** [comment]
Made the changes you suggested and it appears to work great! Take a look when you get a chance.

**Pull request merged**